### DISCUSSION TITLE
Flash Builder IDE's Future

### DISCUSSION THREAD

**Discussion by Shaunmax** [airsdk/Adobe-Runtime-Support#1461] [General]
Hello @ajwfrost,

Could you please tell us about Flash Builder IDE ? Seems like Adobe has stopped its development & its no longer supported on latest Mac OSes. We would like to know, if Harman have any plans to support it or any other IDE in future? 

Thanks
Shaun

**Comment by 2jfw** [comment]
FlashBuilder has reached EOL nearly 10 years ago - you should move ahead and switch to a different IDE, like IntelliJ, Moonshine or VisualStudio Code.

Have a look at https://airsdk.dev/docs/basics/setup-an-editor

Also, telling you from a personal side, you will _not_ regret switching to a more modern IDE - it's like if you would switch from Windows XP to Windows 10

**Reply by marchbold** [reply]
Could not agree more, the day we switched to intellij was like a breath of fresh air, and that was many years ago...

**Reply by marchbold** [reply]
I'd suggest harman invest in one of the newer ide's than trying to revive flash builder. Eclipse is a heavy IDE to start with

**Reply by Shaunmax** [reply]
> FlashBuilder has reached EOL nearly 10 years ago

Really, according to wikipedia : 
![image](https://user-images.githubusercontent.com/16576562/146339373-e84b825f-8156-40ea-9205-7d29f5d24650.png)

@2jfw I have been using IntelliJ for the past two years and couldn't agree more with you. But for starters, IntelliJ is costly and other IDEs can give a lot of hick ups, just to print "Hello World". I have used Moonshine but its dead slow... I have not used FDT, so can't really comment on it. What about Flash Develop, I dont see any releases after 2018 and the Mac version never arrived....

I am curious to know what's in Harman's long term plan with AIR. A dedicated IDE support would definitely add more life to it and attract more indie developers to this platform.

> I'd suggest harman invest in one of the newer ide's than trying to revive flash builder. Eclipse is a heavy IDE to start with

I agree, instead of creating something new, its better to support an existing IDE.

**Reply by 2jfw** [reply]
> > FlashBuilder has reached EOL nearly 10 years ago
> 
> Really, according to wikipedia : ![image](https://user-images.githubusercontent.com/16576562/146339373-e84b825f-8156-40ea-9205-7d29f5d24650.png)

Just read it like "EOL" was a metapher for it no longer having received significant features/improvements 😉🤓

**Reply by toysonlocation** [reply]
I moved to IntelliJ after I borked an FB 4.7 install on Mojave and could not get it running again. Haven't looked back. Setting up an AIR project is pretty quick, and I can be working in minutes. I now also do all my Web development with IntelliJ. I've tried VS Code but the keybindings are just too familiar in IntelliJ. In particular, navigate-to-last-edit-location (even across files). I haven't found another IDE with an equivalent. It took me a few YEARS to get used to paying a subscription fee for it, but now I just frown a little and hit submit. They do keep it in good shape.

**Comment by ajwfrost** [comment]
We wondered initially about Flash Builder - it's still used by a surprising number of companies. But it was licensed as part of Creative Cloud which meant commercially it wasn't going to be easy for us to start separating out, and it is very 'legacy' and old...  plus it's a very different proposition to taking on the AIR runtime/SDK, we didn't have the source code for it, and our license with Adobe mandates certain terms for software distribution which aren't really applicable to an IDE (although we've since had some updates to this - and are further updating it currently - so that we can support Scout etc...)

So I would agree with the comments above: best to move to another platform. Personally I have tended to use FlashDevelop as it's the easiest (free) one I found for setting up a new project, but more recently of course there's Moonshine and VS Code with the extensions that Josh has created, which is very compelling. Plus of course, being able to set up a new project via the [AIR Package Manager](https://github.com/airsdk/apm/wiki) is a big advantage..

**Reply by Shaunmax** [reply]
> FlashDevelop

Unfortunately, its only on Windows and if you are supporting iOS/Mac builds then its going to be a big burden.

> VS Code with the extensions that Josh has created,

Is a good proposition, but needs a good UI interface for project structuring and edit configurations similar to IntelliJ.
